packages/dd-trace/src/plugins/util/ip_extractor.js

```diff
@@ -73,17 +92,48 @@ function findFirstIp (str) {
 
     // TODO: strip port and interface data ?
 
-    const type = net.isIP(chunk)
-    if (!type) continue
-
-    if (!privateIPMatcher.check(chunk, type === 6 ? 'ipv6' : 'ipv4')) {
+    const currentIpInfo = extractPublicOrPrivateIp(chunk)
+    if (currentIpInfo.public) {
       // it's public, return it immediately
       result.public = chunk
       break
     }
 
     // it's private, only save the first one found
-    if (!result.private) result.private = chunk
+    if (currentIpInfo.private && !result.private) result.private = chunk
```

Comment:
actually even better, if you just put back the net.isIP() back here, you can avoid creating all these objects, and just call "isPublicIP()"
```suggestion
    const type = net.isIP(chunk)
    if (!type) continue

    if (isPublicIp(chunk, type)) {
      // it's public, return it immediately
      result.public = chunk
      break
    }

    // it's private, only save the first one found
    if (!result.private) result.private = chunk
```